interoperator/pkg/controller/multiclusterdeploy/provisioner/provisioner_controller.go

```diff
@@ -0,0 +1,433 @@
+// a Deployment as an example
+/* Functions of this method
+1. Add watches on resources in target sfcluster
+2. Get target cluster client
+3. Get statefulset instance deployed in master cluster
+4. Register SF CRDs in target cluster
+5. Namespace creation in target cluster
+6. SFCluster deploy in target cluster
+7. Kubeconfig secret in target cluster
+8. Deploy provisioner in target cluster
+9. Create clusterrolebinding in target cluster
+*/
+func (r *ReconcileProvisioner) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// Fetch the SFCluster
+	clusterInstance := &resourcev1alpha1.SFCluster{}
+	err := r.Get(context.TODO(), request.NamespacedName, clusterInstance)
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			// Object not found, return.
+			err = watchmanager.RemoveCluster(request.Name)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+			return reconcile.Result{}, nil
+		}
+		log.Error(err, "Failed to get SFCluster...", "clusterId", request.NamespacedName.Name)
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+	clusterID := clusterInstance.GetName()
+	log.Info("reconciling cluster", "clusterID", clusterID)
+
+	// Get targetClient for targetCluster
+	targetClient, err := r.clusterRegistry.GetClient(clusterID)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Get statefulSetInstance for provisioner
+	statefulSetInstance := r.provisioner.GetStatefulSet()
+
+	// 1. Register sf CRDs
+	err = r.registerSFCrds(clusterID, targetClient)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	err = watchmanager.AddCluster(clusterID)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// 2. Create/Update Namespace in target cluster for provisioner
+	namespace := statefulSetInstance.GetNamespace()
+	err = r.reconcileNamespace(namespace, clusterID, targetClient)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// 3. Creating/Updating sfcluster in target cluster
+	err = r.reconcileSfClusterCrd(clusterInstance, clusterID, targetClient)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// 4. Creating/Updating kubeconfig secret for sfcluster in target cluster
+	err = r.reconcileSfClusterSecret(request.NamespacedName.Name, namespace, clusterInstance, clusterID, targetClient)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// 5. Create Statefulset in target cluster for provisioner
+	err = r.reconcileStatefulSet(statefulSetInstance, clusterID, targetClient)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// 6. Deploy cluster rolebinding
+	err = r.reconcileClusterRoleBinding(namespace, clusterID, targetClient)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileProvisioner) registerSFCrds(clusterID string, targetClient client.Client) error {
+	SFCrdNames := []string{
+		"sfplans.osb.servicefabrik.io",
```

Comment:
constant ?

Reply:
As discussed, getting it from the discovery client will be the way to go and we would take it in a separate PR.